pkgs/applications/networking/p2p/freenet/default.nix

```diff
@@ -15,46 +15,56 @@ let
     sha256 = "109zn9w8axdkjwhkkcm2s8dvib0mq0n8imjgs3r8hvi128cjsmg9";
   };
   version = "build01470";
-in
-stdenv.mkDerivation {
-  name = "freenet-${version}";
 
+  freenet-jars = stdenv.mkDerivation {
+    name = "freenet-jars-${version}";
 
-  src = fetchgit {
-    url = https://github.com/freenet/fred;
-    rev = "refs/tags/${version}";
-    sha256 = "1b6e6fec2b9a729d4a25605fa142df9ea42e59b379ff665f580e32c6178c9746";
-  };
+    src = fetchgit {
+      url = https://github.com/freenet/fred;
+      rev = "refs/tags/${version}";
+      sha256 = "1b6e6fec2b9a729d4a25605fa142df9ea42e59b379ff665f580e32c6178c9746";
+    };
```

Comment:
What I've done here is moved a "base" derivation into the `let` block, and the main derivation is a wrapper around that. I've since figured out a way I can merge them into one derivation again by using `substituteAll` from stdenv directly, which I will probably do this evening.